numpy/core/src/multiarray/abstractdtypes.c

```diff
@@ -0,0 +1,168 @@
+#define PY_SSIZE_T_CLEAN
+#include <Python.h>
+#include "structmember.h"
+
+
+#define NPY_NO_DEPRECATED_API NPY_API_VERSION
+#define _MULTIARRAYMODULE
+#include "numpy/ndarraytypes.h"
+#include "numpy/arrayobject.h"
+
+#include "abstractdtypes.h"
+#include "array_coercion.h"
+#include "common.h"
+
+
+static PyArray_Descr *
+discover_descriptor_from_pyint(
+        PyArray_DTypeMeta *NPY_UNUSED(cls), PyObject *obj)
+{
+    assert(PyLong_Check(obj));
+    /*
+     * We check whether long is good enough. If not, check longlong and
+     * unsigned long before falling back to `object`.
+     */
+    long long value = PyLong_AsLongLong(obj);
+    if (error_converting(value)) {
+        PyErr_Clear();
+    }
+    else {
+        if (NPY_MIN_LONG <= value && value <= NPY_MAX_LONG) {
+            return PyArray_DescrFromType(NPY_LONG);
```

Comment:
Its handled by the `error_converting` path above. That should probably actually use some `AndOverflow` API to avoid creating and discoarting the error. But right now this code should be a pretty straight forward move only.

Reply:
thanks for the clarification. i guess this hits a bunch of times before its called for 18446744073709551614 (not completely sure why), and i guess i was looking at the call for the wrong obj and that probably confused me. Agree looks pretty straightforward otherwise, and probably should be okay with the ``AndOverflow`` API